### DISCUSSION TITLE
How to rotate 2d depiction?

### DISCUSSION THREAD

**Discussion by khoroshyy** [rdkit/rdkit#4939] [Q&A] [answered]
Is there any way to rotate 2D depiction?
I am not talking about alignment with other structure, more like flipping viewing direction and rotation.
Thanks.
Petro.

**Comment by khoroshyy** [comment]
> Is there any way to rotate 2D depiction?
> I am not talking about alignment with other structure, more like flipping viewing direction and rotation.
> Thanks.
> Petro.

I have found the rotate option, but still no clue how switch to the view from  another side of the screen...

**Comment by dehaenw** [comment, marked as answer]
hello @khoroshyy,
one possible fix to view from the other side of the screen is to apply a 180 deg rotation about the y axis using rdmoltransforms. see below:
```python
import rdkit
from rdkit import Chem
from rdkit.Chem import rdDepictor,rdMolTransforms
import numpy as np
mol = Chem.MolFromSmiles("C([C@@H]1[C@H]([C@@H]([C@H](C(O1)O)O)O)O)O") #glucose
rdDepictor.Compute2DCoords(mol,canonOrient=False) #needed to have one 2d "conformer"
display(mol)
rot_y = np.array([[-1.,0.,0.,0.],[0.,1.,0.,0.],[0.,0.,-1.,0.],[0.,0.,0.,1.]]) # rotate 180 about y axis
rdMolTransforms.TransformConformer(mol.GetConformer(0), rot_y)
display(mol)
```
giving:
![image](https://user-images.githubusercontent.com/66372095/150984319-2e2adff7-7af4-4cfa-8481-60ea04f9d65a.png)

i'm not sure if this is the right way to do it, but this is one way to flip the plane.

**Reply by khoroshyy** [reply]
@dehaenw Thanks. Works like a charm.
